core/templates/pages/exploration-player-page/services/prediction-algorithm-registry.service.ts

```diff
@@ -65,7 +65,7 @@ export class PredictionAlgorithmRegistryService {
       this.algorithmIdPredictionServiceMapping.set(algorithmId, new Map());
     }
     this.algorithmIdPredictionServiceMapping.get(algorithmId)
-      .set(dataSchemaVersion, service);
+      ?.set(dataSchemaVersion, service);
```

Comment:
This is relly confusing, can you try to reformat for better readability?

Reply:
Updated